File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int size = nums.size();
        int count = 0;
        int can = 0; // Declare 'can' outside of the loop
        
        for (int i = 0; i < size; i++) {
            if (count == 0) {
                can = nums[i]; // Update the candidate
            }
            if (can == nums[i]) { // Compare with current element
                count++;
            } else {
                count--;
            }
        }
        return can; // Return the majority element candidate
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        int count=1;
        int max=0;
        int val=nums[0];
        for(int i=0;i<n-1;i++){
            if(nums[i]==nums[i+1]){
                count++;
            }
            else{
                count=1;
            }
            if(max<count){
                max=count;
                val=nums[i];
            }
        }
        return val;
    }
}